### PR TITLE
Fix #3318 - Suppress spinner in parallel runs in CI

### LINKED ISSUE #3318: Set --parallel-no-spinner if in CI
## What's the problem this feature will solve?

`--parallel-no-spinner` is fairly required for CI runs to prevent the spinner from messing the output. It seems reasonable to set the flag on by default if CI is detected.

## Describe the solution you'd like

Set the `--parallel-no-spinner` default based on `is_ci()` utility.

## Alternative Solutions

No alternatives.

## Caveats
There's no `--parallel-spinner` which would force the spinner if it was disabled by the CI detection. `TOX_PARALLEL_NO_SPINNER=0` should still work in those (hopefully rare) cases.

### PR DESCRIPTION
**Pull request by ziima**
Suppress spinner in parallel run in CI (not legacy though).

### CODE REVIEW

**Review by gaborbernat — APPROVED** [review]

File: tests/session/cmd/test_parallel.py [hunk 1]
```diff
@@ -186,6 +187,19 @@ def test_parallel_no_spinner(tox_project: ToxProjectCreator) -> None:
     )
 
 
+def test_parallel_no_spinner_ci(tox_project: ToxProjectCreator) -> None:
+    """Ensure spinner is disabled by default in CI."""
+    with mock.patch.object(parallel, "execute") as mocked, mock.patch.dict(os.environ, {"CI": "1"}):
```
Comment (gaborbernat) [inline tests/session/cmd/test_parallel.py:192]: Use pytest-mock instead

**Review by gaborbernat — CHANGES_REQUESTED** [review]

File: tests/session/cmd/test_parallel.py (hunk 1 above)
Comment (ziima) [reply tests/session/cmd/test_parallel.py]: Fixed

Comment (ziima) [reply tests/session/cmd/test_parallel.py:192]: OK, I had to rolled back. It seems pytest-mock has some issue with `os.environ`. Also `unittest.mock` is used to modify `os.environ` throughout tox tests.

Comment (gaborbernat) [reply tests/session/cmd/test_parallel.py:192]: I strongly urge you to reconsider. For patching the operating system environment variables, you should use the built-in pytest monkeypatch fixture.

Comment (gaborbernat) [reply tests/session/cmd/test_parallel.py:192]: I will not accept any pull request that doesn't use the already practiced tools and introduces a new way of doing the same thing.

**Comment by ziima** [comment]
Thanks for fixing the PR, I didn't get back to that yet.

**Pull request merged**